rewrite-java-test/src/test/java/org/openrewrite/java/ReplaceStringLiteralWithConstantTest.java

```diff
@@ -19,103 +19,253 @@
+            }
+            """,
+            """
+            import org.openrewrite.java.ReplaceStringLiteralWithConstantTest;
+            
+            class Test {
+                Object o = ReplaceStringLiteralWithConstantTest.EXAMPLE_STRING_CONSTANT;
+            }
             """
-              package com.abc;
-              class A {
-                  String v = "newValue";
-                  private String method() {
-                      return "newValue";
-                  }
-              }
-              """,
-            """
-              package com.abc;
-
-              import com.constant.B;
-
-              class A {
-                  String v = B.VAR;
-                  private String method() {
-                      return B.VAR;
-                  }
-              }
-              """
           )
         );
     }
+
+    @Test
+    void replaceStringLiteralWithConstantValueWhenLiteralValueIsConfiguredNull() {
+        rewriteRun(
+          spec -> spec.recipe(new ReplaceStringLiteralWithConstant(null, EXAMPLE_STRING_FQN)),
+          java(
+            """
+            class Test {
+                Object o = "Hello World!";
+            }
+            """,
+            """
+            import org.openrewrite.java.ReplaceStringLiteralWithConstantTest;
+            
+            class Test {
+                Object o = ReplaceStringLiteralWithConstantTest.EXAMPLE_STRING_CONSTANT;
+            }
+            """
+          )
+        );
+    }
+
+    @Test
+    void replaceStringLiteralWithLiteralValueWhenLiteralValueIsConfiguredEmpty() {
+        rewriteRun(
+          spec -> spec.recipe(new ReplaceStringLiteralWithConstant("", EXAMPLE_STRING_FQN)),
+          java(
+            """
+            class Test {
+                Object o = "";
+            }
+            """,
+            """
+            import org.openrewrite.java.ReplaceStringLiteralWithConstantTest;
+            
+            class Test {
+                Object o = ReplaceStringLiteralWithConstantTest.EXAMPLE_STRING_CONSTANT;
+            }
+            """
+          )
+        );
+    }
+
+    @Test
+    void replaceStringLiteralWithLiteralValueWhenLiteralValueIsConfiguredBlank() {
+        rewriteRun(
+          spec -> spec.recipe(new ReplaceStringLiteralWithConstant(" ", EXAMPLE_STRING_FQN)),
+          java(
+            """                
+            class Test {
+                Object o = " ";
+            }
+            """,
+            """
+            import org.openrewrite.java.ReplaceStringLiteralWithConstantTest;
+            
+            class Test {
+                Object o = ReplaceStringLiteralWithConstantTest.EXAMPLE_STRING_CONSTANT;
+            }
+            """
+          )
+        );
+    }
+
+    public static String EXAMPLE_STRING_FQN = ReplaceStringLiteralWithConstantTest.class.getName() + ".EXAMPLE_STRING_CONSTANT";
+    public static String EXAMPLE_STRING_CONSTANT = "Hello World!";
```

Comment:
I'd pull these fields up to the top; reading the code I first thought they might be references elsewhere before finding them here. When put at the top there's no such confusion.

Reply:
You are right!